umibukela/healthe/views.py

```diff
@@ -67,3 +67,18 @@ def stats(request):
     end = date.today() - timedelta(weeks=4)
     rows = get_submissions(start, end, 'json', 'start')
     return JsonResponse(stockout_stats(start, end, rows))
+
+@cache_page(3600)
+def year_report(request):
+    """ Provide full year reports
+    for http:s//www.health-e.org.za/medicine-stockouts/ page
+    """
+    year = request.GET.get('year')
+    start_date = arrow.get('%s-01-01' % year).date()
+    end_date = arrow.get('%s-12-31' % year).date()
```

Comment:
We need to be careful to honour the agreement Health-E has with the dept not to publicly release data until a month after it has been captured. `end_date` should be no later than 4 weeks before today. eg.

```python
cutoff = date.today() - timedelta(weeks=4)
if end_date > cutoff:
    end_date = cutoff
```